libs/providers/flagsmith/src/lib/type-factory.ts

```diff
@@ -0,0 +1,61 @@
+import type { FlagValue, JsonValue } from '@openfeature/server-sdk';
+
+export type FlagType = 'string' | 'number' | 'object' | 'boolean';
+
+const toNumber = (value: unknown): number | undefined => {
+  if (typeof value === 'number' && Number.isFinite(value)) return value;
+  const parsed = parseFloat(value as string);
+  return Number.isFinite(parsed) ? parsed : undefined;
+};
+
+const toBoolean = (value: unknown): boolean | undefined => {
+  if (typeof value === 'boolean') return value;
+  // I added this one in case true/false is passed as a number: wdyt ?
+  if (typeof value === 'number') {
+    return value !== 0;
+  }
+  if (typeof value === 'string') {
+    const lower = value.toLowerCase();
+    if (lower === 'true') return true;
+    if (lower === 'false') return false;
+  }
+  return undefined;
```

Comment:
I kept it very wide so truthy/falsy strings and numbers are considered booleans. Are we ok with this?

Reply:
So you mean that `evalueBooleanResult` will return `true` if the value is e.g. `"foo"`?